packages/integrations/src/download-client/aria2/aria2-integration.ts

```diff
@@ -0,0 +1,181 @@
+import path from "path";
+
+import { fetchWithTrustedCertificatesAsync } from "@homarr/certificates/server";
+
+import type { DownloadClientJobsAndStatus } from "../../interfaces/downloads/download-client-data";
+import { DownloadClientIntegration } from "../../interfaces/downloads/download-client-integration";
+import type { DownloadClientItem } from "../../interfaces/downloads/download-client-items";
+import type { Aria2Download, Aria2GetClient } from "./aria2-types";
+
+export class Aria2Integration extends DownloadClientIntegration {
+  public async getClientJobsAndStatusAsync(): Promise<DownloadClientJobsAndStatus> {
+    const client = this.getClient();
+    const keys: (keyof Aria2Download)[] = [
+      "bittorrent",
+      "uploadLength",
+      "uploadSpeed",
+      "downloadSpeed",
+      "totalLength",
+      "completedLength",
+      "files",
+      "status",
+      "gid",
+    ];
+    const [actives, waitings, stoppeds, globalstats] = await Promise.all([
+      client.tellActive(),
+      client.tellWaiting(0, 1000, keys),
+      client.tellStopped(0, 1000, keys),
+      client.getGlobalStat(),
+    ]);
```

Comment:
I would suggest that we don't load 1000 entries as max here and instead something like 50 / 100 to not fill the memory to much (all data fetched is cached in redis)

Reply:
Can't we use a parameter for how many to fetch?
Also, unless we load binaries into the memory, we should be fine even when loading thousands of entries, no?
As a comparison, 100 paragraphs of the lorem ipsum text is only 0.04MB - which should be easily to handle by Redis.
See https://redis.io/docs/latest/operate/rs/databases/memory-performance/memory-limit/

Reply:
Sounds like a good idea 👍🏼